### PR TITLE
Convert theming APIs to CsWin32

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Creates some simple helper methods in `VisualStyle` classes to reduce code duplication. Uses `OpenThemeDataScope` in more places. Adds `HWNDInternal` property to `Control`.

`ThemeHandle` can (and should) likely be removed. Creating a simple cache collection that has a finalizer to close all of the handles would allow removing it.

Note that currently `HTHEME` isn't defined.  https://github.com/microsoft/win32metadata/issues/1273

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/7855)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/VisualStyleInformation.cs [hunk 1]
```diff
@@ -242,77 +162,41 @@ public static int MinimumColorDepth
...
-                {
-                    if (t_visualStyleRenderer is null)
-                    {
-                        t_visualStyleRenderer = new VisualStyleRenderer(VisualStyleElement.Window.Caption.Active);
-                    }
-                    else
-                    {
-                        t_visualStyleRenderer.SetParameters(VisualStyleElement.Window.Caption.Active);
-                    }
+                PInvoke.GetThemeSysInt(
```
[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/VisualStyleInformation.cs:165]: Yeah, missed that. Thanks :)

File: src/System.Windows.Forms.Primitives/src/Windows/Win32/PInvoke.OpenThemeDataScope.cs [hunk 2]
```diff
@@ -2,40 +2,42 @@
...
         {
-            public IntPtr HTheme { get; }
+            public nint HTheme { get; }
 
             /// <summary>
-            ///  Opens the requested theme data using <see cref="OpenThemeData(IntPtr, string)"/>.
+            ///  Opens the requested theme data using <see cref="OpenThemeData(HWND, string)"/>.
             /// </summary>
-            public OpenThemeDataScope(IntPtr hwnd, string pszClassList)
+            public OpenThemeDataScope(HWND hwnd, string pszClassList, bool throwOnError = false)
```
Comment (RussKie) [inline src/System.Windows.Forms.Primitives/src/Windows/Win32/PInvoke.OpenThemeDataScope.cs:29]: `throwOnError` is not used

File: src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/VisualStyleRenderer.cs [hunk 3]
```diff
@@ -625,10 +602,9 @@ public unsafe Size GetPartSize(IDeviceContext dc, Rectangle bounds, ThemeSizeTyp
 
             // Valid values are 0x0 to 0x2
             SourceGenerated.EnumValidator.Validate(type, nameof(type));
-
+
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/VisualStyleRenderer.cs:605]: [nit] Stray spaces

File: src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/VisualStyleRenderer.cs [hunk 4]
```diff
@@ -650,12 +626,12 @@ public Point GetPoint(PointProperty prop)
         public unsafe Padding GetMargins(IDeviceContext dc, MarginProperty prop)
         {
             ArgumentNullException.ThrowIfNull(dc);
-
+
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/VisualStyleRenderer.cs:629]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/VisualStyleRenderer.cs [hunk 5]
```diff
@@ -710,10 +705,10 @@ public unsafe Rectangle GetTextExtent(IDeviceContext dc, Rectangle bounds, strin
         public TextMetrics GetTextMetrics(IDeviceContext dc)
         {
             ArgumentNullException.ThrowIfNull(dc);
-
+
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/VisualStyles/VisualStyleRenderer.cs:708]: ditto

File: src/System.Windows.Forms.Primitives/src/Windows/Win32/PInvoke.OpenThemeDataScope.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Primitives/src/Windows/Win32/PInvoke.OpenThemeDataScope.cs:29]: Whoops :) I looked at moving the throw in here and back-tracked. Good catch.

**Review by dreddy-work — DISMISSED** [review]
rest LGTM.

**Review by lonitra — APPROVED** [review]
Looks good!

**Comment by elachlan** [comment]
This has conflicts after merge of other cswin32 PR.

**Review by RussKie — APPROVED** [review]

**Pull request merged**